test/gtest/uct/ib/test_cq_moderation.cc

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+extern "C" {
+#include <uct/api/uct.h>
+}
+#include <uct/uct_test.h>
+#include <ucs/time/time.h>
+#include <poll.h>
+#include <infiniband/verbs.h>
+
+#define TEST_PERIOD (3ul * UCS_USEC_PER_SEC) /* wait for 3 secs to get statistics */
+#define NSEC_PER_USEC (UCS_NSEC_PER_SEC / UCS_USEC_PER_SEC)
+
+#define MODERATION_PERIOD 1000 /* usecs */
+#define EVENT_LIMIT (10 * TEST_PERIOD / MODERATION_PERIOD / NSEC_PER_USEC)
+#define MAX_REPEATS 1000
+
+#define _STR(_arg) #_arg
+#define TO_STR(_val) _STR(_val)
+
+class test_uct_cq_moderation : public uct_test {
+protected:
+
+    void init() {
+        uct_test::init();
+
+        if (!(GetParam()->tl_name == "rc" || GetParam()->tl_name == "rc_mlx5" ||
+              GetParam()->tl_name == "ud" || GetParam()->tl_name == "ud_mlx5" ||
+              GetParam()->tl_name == "dc" || GetParam()->tl_name == "dc_mlx5")) {
```

Comment:
don't understand, looks like all IB transports are enabled?

Reply:
artefact macro (some DC tests failed before... fixed)

Reply:
so let's remove?

Reply:
comment is removed.
or you mean remove this evaluation for transport name?

Reply:
yes, remove the transport name check

Reply:
ok, got your point.
updated transport check: used _IB_ macro to test class instantiate